apps/core/scenes/resultDetail/ResultDetailInner.js

```diff
@@ -31,12 +38,30 @@ class ResultDetailInner extends React.Component<Props> {
   }
 
   componentDidUpdate(prevProps) {
-    if (
-      this.props.data?.checkItinerary?.isChecked !==
-      prevProps.data?.checkItinerary?.isChecked
-    ) {
-      this.stopItineraryCheck();
-      this.tryStartItineraryCheck();
+    const { passengers, data, relay, bookingToken } = this.props;
+    const numPassengers = passengers.length;
+    const passengersChanged = prevProps.passengers.length !== numPassengers;
+    const checkedChanged =
+      data?.checkItinerary?.isChecked !==
+      prevProps.data?.checkItinerary?.isChecked;
+
+    if (checkedChanged || passengersChanged) {
```

Comment:
We need to find better architecture because let's say a user search flight for three passengers at the start. When a user is on the result-detail page first checkItinerary run for 3 passengers. But when a user adds a first passenger in to form, we call the checkItinerary with one, if itinerary will be not checked successfully we will call `this.tryStartItineraryCheck()` until is checked, but here is a problem because of the `tryStartItineraryCheck` call `relay.refetch()` with params from search so it will be for 3 passengers now.
This probably won't be a big problem but what if the user will be searched itinerary just for one passenger and decided added 4 passengers? The tryStartItineraryCheck will be called with one but we need to check if a flight is valid for 4 passengers. 
And the price should change also.

Reply:
We should somehow simplify this flow. 🤔

Reply:
Yes. This `tryStartItineraryCheck` is weird ahahaha, we should maybe revisit it in another PR? or this one, like: `Fix passengers itinerary check`

Reply:
Currently, this PR doesn't solve the problem with check itinerary. These two checks with different params just lead to inconsistency. 

I am trying to figure out a simple solution on how to do deal with that.